src/api/app/models/kiwi/image.rb

```diff
@@ -48,43 +49,51 @@ def self.build_from_xml(xml_string, md5)
     order = 1
     repositories = xml["repository"]
     repositories = [xml["repository"]] if xml["repository"].is_a?(Hash)
-    repositories.each do |repository|
-      attributes = {
-        repo_type:   repository['type'],
-        source_path: repository['source']['path'],
-        priority:    repository['priority'],
-        order:       order,
-        alias:       repository['alias'],
-        replaceable: repository['status'] == 'replaceable',
-        username:    repository['username'],
-        password:    repository['password']
-      }
-      attributes['imageinclude'] = repository['imageinclude'] == 'true' if repository.key?('imageinclude')
-      attributes['prefer_license'] = repository['prefer-license'] == 'true' if repository.key?('prefer-license')
-
-      new_image.repositories.build(attributes)
-      order += 1
+
+    if repositories
+      new_image.use_project_repositories = true if repositories.map { |repository| repository['source']['path'] }.include?('obsrepositories:/')
+
+      repositories.reject{ |repository| repository['source']['path'] == 'obsrepositories:/' }.each do |repository|
+        attributes = {
+          repo_type:   repository['type'],
+          source_path: repository['source']['path'],
+          priority:    repository['priority'],
+          order:       order,
+          alias:       repository['alias'],
+          replaceable: repository['status'] == 'replaceable',
+          username:    repository['username'],
+          password:    repository['password']
+        }
+        attributes['imageinclude'] = repository['imageinclude'] == 'true' if repository.key?('imageinclude')
+        attributes['prefer_license'] = repository['prefer-license'] == 'true' if repository.key?('prefer-license')
+
+        new_image.repositories.build(attributes)
+        order += 1
+      end
     end
+
     package_groups = xml["packages"]
     package_groups = [xml["packages"]] if xml["packages"].is_a?(Hash)
-    package_groups.each do |package_group_xml|
-      attributes = {
-        kiwi_type:    package_group_xml['type'],
-        profiles:     package_group_xml['profiles '],
-        pattern_type: package_group_xml['patternType']
-      }
-      package_group = Kiwi::PackageGroup.new(attributes)
-      package_group_xml['package'].each do |package|
+    if package_groups
```

Comment:
Why is this necessary now? I think it should always be an array, no?

Reply:
@ChrisBr, It was always necessary because `package_groups = xml["packages"]` return `nil` if there not exist any packages. The same happens with repositories.